cli/test/test_runner_slurm.py

```diff
@@ -1,133 +1,241 @@
+            self.slurm_runner._tail_output('slurm-x.out'), 0)
+        self.assertEqual(len(self.slurm_runner._out_stream_pid), 1)
 
-    def test_generate_script(self):
-        cmd = " ".join(["docker", "version"])
-        job_script = os.path.join(self.repo, 'script.sh')
-        SlurmRunner.generate_script(cmd, 'sample_job', job_script)
-        with open(job_script, 'r') as f:
+    def test_stop_running_tasks(self):
+        self.Popen.set_command('scancel --name job_a', returncode=0)
+        self.slurm_runner._spawned_jobs.add('job_a')
+        self.slurm_runner.stop_running_tasks()
+        self.assertEqual(
+            call.Popen(
+                ['scancel', '--name', 'job_a'],
+                cwd=os.getcwd(),
+                env=None, preexec_fn=os.setsid, stderr=-2, stdout=-1,
+                universal_newlines=True) in self.Popen.all_calls, True)
+
+    @replace('popper.runner_slurm.os.kill', mock_kill)
+    def test_submit_batch_job(self, mock_kill):
+        self.Popen.set_command(
+            'sbatch --wait '
+            '--job-name popper_sample_123abc '
+            '--output /tmp/popper/slurm/popper_sample_123abc.out '
+            '/tmp/popper/slurm/popper_sample_123abc.sh',
+            returncode=0)
+        self.Popen.set_command(
+            'tail -f /tmp/popper/slurm/popper_sample_123abc.out', returncode=0)
+        config = PopperConfig(workspace_dir='/w')
+        config.wid = "123abc"
+        step = {"name": "sample"}
+        slurm_runner = SlurmRunner(config=config)
+        slurm_runner._submit_batch_job(["ls -la"], step)
+        with open("/tmp/popper/slurm/popper_sample_123abc.sh", 'r') as f:
             content = f.read()
-        self.assertEqual(content, "#!/bin/bash\ndocker version")
 
-    def test_touch_log_files(self):
-        out_file = os.path.join(self.repo, 'slurm-x.out')
-        err_file = os.path.join(self.repo, 'slurm-x.err')
-        self.slurm_runner.touch_log_files(out_file, err_file)
-        self.assertEqual(os.path.exists(out_file), True)
-        self.assertEqual(os.path.exists(err_file), True)
-
-    def test_cancel_job(self):
-        self.Popen.set_command('scancel --name job_a', returncode=0)
-        SlurmRunner.spawned_jobs.add('job_a')
-        SlurmRunner.cancel_job()
+        self.assertEqual(content, "#!/bin/bash\nls -la")
+        self.assertEqual(len(slurm_runner._spawned_jobs), 0)
+        self.assertEqual(slurm_runner._out_stream_thread.is_alive(), False)
+
+    @replace('popper.runner_slurm.os.kill', mock_kill)
+    def test_submit_job_failure(self, mock_kill):
+        self.Popen.set_command(
+            'sbatch --wait --job-name popper_1_123abc '
+            '--output /tmp/popper/slurm/popper_1_123abc.out '
+            '/tmp/popper/slurm/popper_1_123abc.sh', returncode=12)
+
+        self.Popen.set_command(
+            'tail -f /tmp/popper/slurm/popper_1_123abc.out',
+            returncode=0)
+        repo = testutils.mk_repo().working_dir
```

Comment:
same comment here regarding the creation of a git repo (we can use `tempfile` instead)